rskj-core/src/main/java/co/rsk/mine/MinerServer.java

```diff
@@ -26,15 +26,28 @@
 public interface MinerServer {
 
     void start();
+    void stop();
+
+    boolean isRunning();
 
     SubmitBlockResult submitBitcoinBlock(String blockHashForMergedMining, co.rsk.bitcoinj.core.BtcBlock bitcoinMergedMiningBlock);
 
+    boolean generatePrivateBlock();
+
+    void setPrivateMining(boolean p);
+
+    void setAutoSwitchBetweenNormalAndPrivateMining(boolean p);
```

Comment:
Why to have change of state?

The private miners generates blocks, at any rate.

All the private mined blocks has a fixed assigned difficulty.

When normal mining pools are active, those blocks are rejected by low difficulty

When normal mining pool is missing, the network switch to low difficulty. Then, these private mined blocks will start to win the block fight.

Considerations:

- The private mined blocks override any delta rule (PoW previous block vs PoW current block)
- The private mined blocks skip Proof of work rule

Reply:
I subscribe this review :-)

Reply:
Also, this approach allows the switch off all private miner nodes, and then, the normal CPU miiners could have a change of produce a block. 

The use of private mining is up to our decision, without further changes to the distributed code.

Reply:
> Why to have change of state?
Because not all miners will be running the fallback mining system. Only some trusted miners. Then these miners will have auto-switch enabled: they can activate fallback mining when the blockchain difficulty is low.

> The private miners generates blocks, at any rate.
This is how it was implemented.

> All the private mined blocks has a fixed assigned difficulty.
This is not how it's implemented because fallback mined blocks can be intermingled  with PoW mined blocks, the difficulty must stay as real as possible. Fallback mined blocks can increase/decrease the difficulty depending on the block timestamp. Fallback miners will always mine blocks at one every 15 seconds (+1 seconds of the average interval) to keep the difficulty at the minimum, when possible.

> When normal mining pools are active, those blocks are rejected by low difficulty
Not that easy, because we have the rule that after 10 minutes difficulty drops. This was necessary in order to create the fallback system as fast as possible. The ProofOfWorkRule does not know the parent block timestamp, and therefore cannot act on the parent block difficulty. 

> When normal mining pool is missing, the network switch to low difficulty. Then, these private mined blocks will start to win the block fight.
This is how it was implemented.